scripts/agent_workflow.py

```
@@ -0,0 +1,219 @@
+    """A safe, user-actionable workflow failure."""
+
+
+@dataclass(frozen=True)
+class WorkspaceIdentity:
+    workspace_key: str
+    repository: str
+    branch: str
+
+
+def _git(cwd: Path, *args: str) -> str | None:
+    result = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)